### DISCUSSION TITLE
From code in <script setup> (vue 3), how to update existing marker created in <template> ?

### DISCUSSION THREAD

**Discussion by chaperot-perso** [vue-leaflet/vue-leaflet#229] [Q&A]
Hello,

I am using vue-leaflet on vue 3 app.
I use composition API.
A marker is declared in <template> <l-map> <l-control> <l-marker>.
From code in <script setup>, I would like to modify the position (latLng) but I do not succeed to get a variable pointing to that object.

. How can I get in <script setup> a value pointing to a marker created in template ?
. How can I create from <script setup> a new marker ? a new polygon ?

Thanks :)

**Comment by mikeu** [comment]
Hi @chaperot-perso, would you be able to share the specific code you're using? I also think something in your original message got dropped, it doesn't seem to say where you are declaring your marker. From the sound of things I think you'll likely want to be using a `ref`, but it will be easier to be sure once we're discussing a minimal working example. Thanks!

**Comment by chaperot-perso** [comment]
Hello Mikeu,

Many thanks for this first answer!

Here is a code to describe the situation I meet:
[LeafletDebug.zip](https://github.com/vue-leaflet/vue-leaflet/files/9383144/LeafletDebug.zip)
The only way I found to get a variable pointing to the marker was to catch evt.target in marderMoveEnd event...
To get it run, I need to move the cursor with mouse, and then I get a variable pointing to it.
How to initialize a variable pointing to the marker without my ugly and dirty trick ?

Another question,
From <script setup>, I wish to add, modify, remove polygons on map.
How can I do that ?

Thanks :)
